### PR TITLE
Azure Developer CLI Post

### PR DESCRIPTION
**Pull request by savannahostrowski**

### CODE REVIEW

**Comment by savannahostrowski** [comment]
@puicchan PTAL - the post is largely finished with the exception of `pipeline config` (waiting until the fix goes in for the release tomorrow).

@austinauth FYI - for the graphic

File: website/blog/2022-09-29/index.md [hunk 1]
```diff
@@ -24,39 +24,124 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
 
 ![](./img/banner.png)
 
 ---
 
+## **A new tool has entered the chat:** What is the Azure Developer CLI (`azd`)?
+The Azure Developer CLI (`azd`) is an new, open source tool that makes it quick and simple for you to move your application from your local development environment to Azure while considering your end-to-end developer workflow. You might be familiar with other CLIs that focus on Infrastructure as Code or scaffolding your application but the Azure Developer CLI does all that and more!
 
-## Section 1
+The Azure Developer CLI commands are simple, high-level and map to core stages in your developer workflow. Think project initialization/creation, build, deploy, repeat!
```
Comment (savannahostrowski) [inline website/blog/2022-09-29/index.md:55]: Insert graphic here.

File: website/blog/2022-09-29/index.md [hunk 2]
```diff
@@ -24,39 +24,124 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
+
+We can also navigate to the Application Dashboard by clicking on the resource group name, where you can set a specific refresh rate for the dashboard, and see usage, reliability, and performance metrics over time.
+![](img/dashboard.png)
+
+I don't know about everyone else but I have spent a ton of time building out similar dashboards. It can be super time-consuming to write all the queries and create the visualizations so this feels like a real time saver.
+
+### CI/CD
+Finally let's talk about setting up CI/CD! This might be my favorite `azd` feature. As I mentioned before, the Azure Developer CLI has a command, `azd pipeline config`, which uses the files in the `.github/` directory to set up a GitHub Action. More than that, if there is no upstream repo, the Developer CLI will actually help you create one.
+
+**@savannah to finish**
```
Comment (savannahostrowski) [inline website/blog/2022-09-29/index.md:131]: Add screenshots after 9/14

**Review by nitya — APPROVED** [review]
Will *wait to** merge this in and update banner etc. later this week. Feel free to send updates up till publication date

**Comment by savannahostrowski** [comment]
@puicchan @jongio @austinauth @rajeshkamal5050 @nitya This is now ready for review. Please take a look folks!

**Comment by puicchan** [comment]
One typo else LGTM.

**Review by puicchan — CHANGES_REQUESTED** [review]

File: website/blog/2022-09-29/index.md [hunk 3]
```diff
@@ -24,39 +24,133 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
+
+When designing the CLI, we wanted the experience to be both flexible and non-magical (no side effects, easy to understand). That being said, we're going to run this all with `azd up` but you could alternatively run a series of three commands and the outcome would be the same - `azd init --template Azure-Samples/todo-python-mongo-swa-func` (to pull the code down to your machine), `azd provision` (to provision infrastructure) and then `azd deploy` (to deploy application code on Azure). Choose your own adventure!
+
+![](img/single-step.png)
+
+So let's walk through it. On running `azd up --template Azure-Samples/todo-python-mongo-swa-func`, I'm prompted for a couple pieces of information as part of the `azd init` process being run under the hood:
+    - **An environment name** - the prefix for the resource group that will be created to hold all Azure resources
+    - **An Azure region** - the Azure location where your resources will be deployed
+    - **An Azure subscription** - the Azure location where your resources will be deployed
```
Comment (rajeshkamal5050) [inline website/blog/2022-09-29/index.md:103]: the Azure account/subscription used for deploying resources

File: website/blog/2022-09-29/index.md [hunk 4]
```diff
@@ -24,39 +24,133 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
+
+### Application development support (code, run, debug, test)
+  - `.devcontainer/` - support for if you're [writing code in a container](https://code.visualstudio.com/docs/remote/containers)
+  - `.vscode/` - support for local development in Visual Studio Code via [launch.json](https://code.visualstudio.com/docs/editor/debugging) (for debugging) and [tasks.json](https://code.visualstudio.com/docs/editor/tasks) (for spinning up the web app for local development)
+  - `src/` - contains all the sample application code which you can modify or swap out for your own application
+  - `tests/` - test for the application, written using [Playwright](https://playwright.dev/)
+
+### Infrastructure as Code (provisioning and deploying infrastructure on Azure, programmatically)
+  - `infra/` - contains all infrastructure as code (IaC) assets written in [Bicep](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/overview?tabs=bicep); includes logic to set up all components we need to set up the application on Azure, wire everything up securely, and monitor application health, performance and usage!
```
Comment (rajeshkamal5050) [inline website/blog/2022-09-29/index.md:85]: callout on support for Terraform?

File: website/blog/2022-09-29/index.md (hunk 3 above)
Comment (savannahostrowski) [reply website/blog/2022-09-29/index.md:103]: Yeah - I copied this from the docs for consistency but I agree. It seems like a mistake that the description for region and sub are the same.

@puicchan Thoughts here?

Comment (puicchan) [reply website/blog/2022-09-29/index.md:103]: In readme, it's 
image

Comment (savannahostrowski) [reply website/blog/2022-09-29/index.md:103]: Ah - my bad. I think we should still consider amending that text. We can chat.

File: website/blog/2022-09-29/index.md [hunk 5]
```diff
@@ -24,39 +24,133 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
+
+
+## **End-to-end support:** Move code to cloud in a single step!
+### If you want to follow along in this section and don't already have the Azure Developer CLI installed, check out [these instructions](https://aka.ms/azure-dev/install-instructions?source=serverless-september) to install it on Windows, Linux or MacOS in your favorite terminal!
+
+So now that we've gone over what this template contains on GitHub, let's pull this template code down to our local machine, set it up for local development, provision the right infrastructure, and deploy the code on Azure in a **single step**.
+
+When designing the CLI, we wanted the experience to be both flexible and non-magical (no side effects, easy to understand). That being said, we're going to run this all with `azd up` but you could alternatively run a series of three commands and the outcome would be the same - `azd init --template Azure-Samples/todo-python-mongo-swa-func` (to pull the code down to your machine), `azd provision` (to provision infrastructure) and then `azd deploy` (to deploy application code on Azure). Choose your own adventure!
```
Comment (jongio) [inline website/blog/2022-09-29/index.md:96]: By using "that being said, " it feels like you are saying there is magic.  You might want to replace it with "Next" or something like that.

File: website/blog/2022-09-29/index.md [hunk 6]
```diff
@@ -24,39 +24,133 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
+
+![](img/single-step.png)
+
+So let's walk through it. On running `azd up --template Azure-Samples/todo-python-mongo-swa-func`, I'm prompted for a couple pieces of information as part of the `azd init` process being run under the hood:
```
Comment (jongio) [inline website/blog/2022-09-29/index.md:100]:
```suggestion
So let's walk through it. On running `azd up -t todo-python-mongo-swa-func`, I'm prompted for a couple pieces of information as part of the `azd init` process being run under the hood:
```

File: website/blog/2022-09-29/index.md [hunk 7]
```diff
@@ -24,39 +24,133 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
+    - **An environment name** - the prefix for the resource group that will be created to hold all Azure resources
+    - **An Azure region** - the Azure location where your resources will be deployed
+    - **An Azure subscription** - the Azure location where your resources will be deployed
+![](img/start-up.png)
+
+Once that information is provided, `azd` will pull down the code from GitHub and create a `.azure/` in the project root that contains all Azure Developer CLI environment information that you just entered. This directory will be important when it comes time to provision and deploy infrastructure in the next step in the `up` process.
```
Comment (jongio) [inline website/blog/2022-09-29/index.md:106]:
```suggestion
Once that information is provided, `azd` will pull down the code from GitHub and create a `.azure/` directory in the project root that contains all Azure Developer CLI environment information that you just entered. This directory will be important when it comes time to provision and deploy infrastructure in the next step in the `up` process.
```

File: website/blog/2022-09-29/index.md [hunk 8]
```diff
@@ -24,39 +24,133 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
+
+The next step here is provisioning. `azd` is running `azd provision` on your behalf and leveraging the IaC assets in the `.infra/` directory in the project. As the tool works to provision, you'll see an output of each resource (name alongside a unique identifier which you can use to reference back to the Azure Portal, if you want)
+![](img/provision-up.png)
+
+Finally, the final step here in running `azd up` is deployment. `azd` is running `azd deploy` and deploying the application code to the resources that we're provisioned in the previous phase of the process. Once this has completed, you'll be able to click on two different endpoint URLs - one for the back-end and one for the front-end. 
+![](img/deploy-up.png)
+
+The back-end endpoint (`service api`) hosts the specification for the API via the `openapi.json` file that's also in the root of the project template. You can explore the endpoints that are available in the web app here.
```
Comment (jongio) [inline website/blog/2022-09-29/index.md:114]: We don't hyphen frontend and backend in our docs.

Comment (jongio) [inline website/blog/2022-09-29/index.md:114]:
```suggestion
The back-end endpoint (`service api`) hosts the specification for the API via the `openapi.yaml` file that's also in the root of the project template. You can explore the endpoints that are available in the web app here. 
```

File: website/blog/2022-09-29/index.md [hunk 9]
```diff
@@ -24,39 +24,133 @@ tags: [serverless-september, 30-days-of-serverless,  azure-container-apps, dapr,
...
+
+The front-end endpoint (`service web`) hosts a fully-fledged and functional ToDo web app with a UI, Cosmos DB for the database and Key Vault for application secrets. This isn't just application hosting. It's really everything you need to be successful and productive, all set up on your behalf by the Azure Developer CLI.
+![](img/frontend.png)
+
+...and that's it! We've successfully deployed our application on Azure! 
+
+But there's more!
+
+## Best practices: Monitoring and CI/CD!
+In my opinion, it's not enough to _just_ set up the application on Azure! I want to know that my web app is performant and serving my users reliably! I also want to make sure that I'm not inadvertently breaking my application as I continue to make changes to it. Thankfully, the Developer CLI also handles all of this via two additional commands - `azd monitor` and `azd pipeline config`.
```
Comment (jongio) [inline website/blog/2022-09-29/index.md:125]:
```suggestion
In my opinion, it's not enough to _just_ set up the application on Azure! I want to know that my web app is performant and serving my users reliably! I also want to make sure that I'm not inadvertently breaking my application as I continue to make changes to it. Thankfully, the Azure Developer CLI also handles all of this via two additional commands - `azd monitor` and `azd pipeline config`.
```

Comment (jongio) [reply website/blog/2022-09-29/index.md:125]: Prob want to add Azure to anywhere you have Developer CLI

**Pull request merged**